src/listen/listener-registry.js

```diff
@@ -842,23 +853,42 @@ module.exports = class ListenerRegistry {
   }
 
   /**
-  * Validates that the pattern is not empty and is a valid regular expression
+  * Converts a pattern to a RegExp object
   *
-  * @param   {SocketWrapper} socketWrapper
-  * @param   {String} pattern
+  * Expects the pattern to be a glob expression unless the `enableRegexListenPatterns` option is
+  * set, in which case it expects a valid regular expression
+  *
+  * returns false if the pattern is empty or is not a valid glob/regular expression
+  *
+  * @param   {SocketWrapper?} socketWrapper
+  * @param   {String}         pattern
   *
   * @returns {RegExp}
   */
-  _validatePattern (socketWrapper, pattern) {
+  _compilePattern (socketWrapper, pattern) {
     if (!pattern) {
       return false
     }
-
-    try {
-      return new RegExp(pattern)
-    } catch (e) {
-      this._onMsgDataError(socketWrapper, e.toString())
-      return false
+    if (this._options.enableRegexListenPatterns === true) {
+      try {
+        return new RegExp(pattern)
+      } catch (e) {
+        if (socketWrapper) {
+          this._onMsgDataError(socketWrapper, e.toString())
+        }
+        return false
+      }
+    } else {
+      // convert glob pattern to RegExp, returns false if pattern is invalid
+      const regExp = minimatch.makeRe(pattern, {
+        noglobstar: true, // disable ** patterns
+        nocomment: true,  // patterns beginning with # shouldn't be ignored
+        noext: true       // disable "extglob" patterns
+      })
+      if (!regExp && socketWrapper) {
+        this._onMsgDataError(socketWrapper, `Invalid glob expression: ${pattern}`)
+      }
+      return regExp
```

Comment:
that's the goal, the work here is clearly not optimal and was intended more as an implementation of the api for listening going forwards